server/dive_tasks/tasks.py

```diff
@@ -545,3 +545,104 @@ def convert_images(self: Task, folderId):
+            if ratio > 600:
+                manager.write(
+                    f"Compression ratio is exceedingly high at {ratio}\n\
+                    Please contact an admin at viame-web@kitware.com if this is a valid zip file"
+                )
+                manager.updateStatus(JobStatus.ERROR)
+                return
+
+            for fileName in listOfFileNames:
+                # contains a sub folder
+                if fileName.endswith('/') and fileName.count('/') == 1:
+                    top_level_folders[fileName.replace('/', '')] = 'flat'
+                # sub folder has a meta.json so it is imported
+                if fileName.endswith(constants.MetaFileName):
+                    root_folder = fileName.replace(f'/{constants.MetaFileName}', '')
+                    top_level_folders[root_folder] = 'dataset'
+                if fileName.endswith('.zip'):
+                    manager.write("Nested Zip files are invalid\n")
+                    manager.updateStatus(JobStatus.ERROR)
+                    return
+                manager.write(f"Extracting: {fileName}\n")
+                zipObj.extract(fileName, f'{_working_directory}')
+        # remove the zip file
+        os.remove(file_name)
+        # validation of files in folder using dive/data
+        created_folder = gc.createFolder(
+            folderId,
+            constants.SourceFolderName,
+            reuseExisting=True,
+        )
+        gc.sendRestRequest(
+            "PUT",
+            f"/item/{str(item['_id'])}?folderId={str(created_folder['_id'])}",
+        )
+
+        if len(top_level_folders) > 0:
+            for top_folder in top_level_folders.keys():
+                base_folder = ''  # default for no subfolder
+                # single flat/dataset import
+                if len(top_level_folders) != 1:
+                    base_folder = top_folder
+                # multiple flat/dataset import
+                if top_level_folders[top_folder] == 'flat':
+                    utils.upload_zipped_flat_media_files(
+                        gc,
+                        manager,
+                        folderId,
+                        Path(f"{_working_directory}/{top_folder}"),
+                        base_folder,
+                    )
+                # multiple flat file exports combined
+                elif top_level_folders[top_folder] == 'dataset':
+                    utils.upload_exported_zipped_dataset(
+                        gc,
+                        manager,
+                        folderId,
+                        Path(f"{_working_directory}/{top_folder}"),
+                        base_folder,
+                    )
+
+        else:  # flat zip file with media and/or annotation data inside
+            utils.upload_zipped_flat_media_files(gc, manager, folderId, Path(_working_directory))
+
+        # remove auxilary folder and metadata if there are sub folders
+        if len(top_level_folders) > 1:
```

Comment:
What's the point of deleting its auxiliary folder?  Seems fine to me to just remove the metadata and move on.

Also, what's the point of removing anything except for DatasetMarker?  That's the only one that matters.

Reply:
It felt weird if you were uploading multiple sub-datasets to have a root folder which contained an Aux folder.  I also didn't want to have a folders lingering around with `type: zip` and `fps` if we did some batch search for a type.  I wanted to behave more like a standard folder with sub datasets that just happens to have a Source folder with the zip in it.